pkg/apply/merge.go

```diff
@@ -76,18 +76,25 @@ func MergeDeploymentForUpdate(current, updated *uns.Unstructured) error {
 	return nil
 }
 
-// MergeServiceForUpdate ensures the clusterip is never written to
+// MergeServiceForUpdate ensures the ClusterIP/IPFamily is never modified
 func MergeServiceForUpdate(current, updated *uns.Unstructured) error {
 	gvk := updated.GroupVersionKind()
 	if gvk.Group == "" && gvk.Kind == "Service" {
 		clusterIP, found, err := uns.NestedString(current.Object, "spec", "clusterIP")
 		if err != nil {
 			return err
 		}
-
 		if found {
 			return uns.SetNestedField(updated.Object, clusterIP, "spec", "clusterIP")
 		}
+
+		ipFamily, found, err := uns.NestedString(current.Object, "spec", "ipFamily")
+		if err != nil {
+			return err
+		}
+		if found {
+			return uns.SetNestedField(updated.Object, ipFamily, "spec", "ipFamily")
```

Comment:
it would be desirable to add a log statement to indicate that the `Service.Spec.ipFamily` was not updated as a result of the merge.

Reply:
The point of not updating it is that it's immutable and so trying to update it would be guaranteed to fail, so logging that we're not updating it doesn't provide any information.

(I guess I didn't totally explain the bug; the problem is that our template `Service` objects don't have `ipFamily` set in them, but once the service is created, it will get a default value of `IPv4`, but the way the code is currently written, we then end up trying to overwrite that with `ipFamily: ""` like in the YAML template, and that results in an error. The fix is to just ignore `ipFamily` on updates, since it can't be changed anyway.)

Reply:
OK, that makes sense. If the field itself is immutable, there is little value. My point was to simply add a DBG log statement saying any updates to the `ipFamily` will be ignored because the field is immutable. But I guess I am OK with it either way.